include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -162,21 +166,32 @@ template <typename Iter, typename ValueType = std::decay_t<typename std::iterato
 using __usm_host_alloc_vec_iter =
     typename std::vector<ValueType, typename sycl::usm_allocator<ValueType, sycl::usm::alloc::host>>::iterator;
 
-// Evaluates to true if the provided type is an iterator with a value_type and if the implementation of a
+// Evaluates to true_type if the provided type is an iterator with a value_type and if the implementation of a
 // std::vector<value_type, Alloc>::iterator can be distinguished between three different allocators, the
 // default, usm_shared, and usm_host. If all are distinct, it is very unlikely any non-usm based allocator
 // could be confused with a usm allocator.
 template <typename Iter>
-constexpr bool __vector_iter_distinguishes_by_allocator_v =
-    !std::is_same_v<__default_alloc_vec_iter<Iter>, __usm_shared_alloc_vec_iter<Iter>> &&
-    !std::is_same_v<__default_alloc_vec_iter<Iter>, __usm_host_alloc_vec_iter<Iter>> &&
-    !std::is_same_v<__usm_host_alloc_vec_iter<Iter>, __usm_shared_alloc_vec_iter<Iter>>;
+struct __vector_iter_distinguishes_by_allocator
+    : std::conjunction<std::negation<std::is_same<__default_alloc_vec_iter<Iter>, __usm_shared_alloc_vec_iter<Iter>>>,
+                       std::negation<std::is_same<__default_alloc_vec_iter<Iter>, __usm_host_alloc_vec_iter<Iter>>>,
+                       std::negation<std::is_same<__usm_host_alloc_vec_iter<Iter>, __usm_shared_alloc_vec_iter<Iter>>>>
+{
+};
+
+template <typename Iter>
+inline constexpr bool __vector_iter_distinguishes_by_allocator_v =
+    __vector_iter_distinguishes_by_allocator<Iter>::value;
+
+template <typename Iter>
+struct __is_known_usm_vector_iter
+    : std::conjunction<__vector_iter_distinguishes_by_allocator<Iter>,
+                       std::disjunction<std::is_same<Iter, oneapi::dpl::__internal::__usm_shared_alloc_vec_iter<Iter>>,
+                                        std::is_same<Iter, oneapi::dpl::__internal::__usm_host_alloc_vec_iter<Iter>>>>
+{
+};
```

Comment:
...then we may reimplement this as
```C++
template <typename Iter>
using __is_known_usm_vector_iter_t =
      std::conjunction<__vector_iter_distinguishes_by_allocator<Iter>,
                       std::disjunction<std::is_same<Iter, oneapi::dpl::__internal::__usm_shared_alloc_vec_iter<Iter>>,
                                        std::is_same<Iter, oneapi::dpl::__internal::__usm_host_alloc_vec_iter<Iter>>>>;
```

Reply:
Do you have an example of a `true/false` trait where we use `_t` suffix?  
My understanding is that the convention is to use `_t` suffix when it is meant to describe a type rather than a binary switch (bool_constant).  I'm looking at the examples `is_execution_policy[_v]`, `__is_sycl_accessor[_v]`, `__is_*_tag[_v]` and following that convention.

Reply:
https://github.com/uxlfoundation/oneDPL/blob/87ad0a609ef250e4a36957a127e0b6d48f021646/include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h#L365
 - but it's without `_t` postfix...
https://github.com/uxlfoundation/oneDPL/blob/87ad0a609ef250e4a36957a127e0b6d48f021646/include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h#L54
 - the same..
 - So I don't insist on `_t` postfix of course, but what is more correct - to use `struct` or template alias here?

Reply:
Adjusted to be an alias.